packages/agreement-lifecycle/src/filters/attributesFilter.ts

```diff
@@ -19,7 +19,17 @@ export const filterVerifiedAttributes = (
       att.verifiedBy.find(
         (v) =>
           v.id === producerId &&
-          (!v.extensionDate || v.extensionDate > new Date())
+          !att.revokedBy.find((revocation) => revocation.id === v.id) &&
+          ((!v.extensionDate && !v.expirationDate) ||
+            (v.extensionDate &&
+              v.expirationDate &&
+              v.extensionDate > new Date() &&
+              v.expirationDate > new Date()) ||
+            (v.extensionDate
+              ? v.extensionDate > new Date()
+              : v.expirationDate
+              ? v.expirationDate > new Date()
+              : false))
```

Comment:
May I suggest another implementation of this function? 

```typescript
export const filterVerifiedAttributes = (
  producerId: TenantId,
  tenantAttributes: TenantAttribute[]
): VerifiedTenantAttribute[] => {
  const isVerificationRevoked = (
    attribute: VerifiedTenantAttribute,
    verifierId: TenantId
  ): boolean =>
    attribute.revokedBy.some((revocation) => revocation.id === verifierId);

  const isVerificationExpired = (
    verifier: VerifiedTenantAttribute["verifiedBy"][0]
  ): boolean => {
    const now = new Date();

    if (verifier.extensionDate && verifier.expirationDate) {
      return verifier.extensionDate <= now && verifier.expirationDate <= now;
    }

    if (verifier.extensionDate) {
      return verifier.extensionDate <= now;
    }

    if (verifier.expirationDate) {
      return verifier.expirationDate <= now;
    }

    return false;
  };

  return tenantAttributes.filter(
    (att): att is VerifiedTenantAttribute =>
      att.type === tenantAttributeType.VERIFIED &&
      att.verifiedBy.some(
        (v) =>
          v.id === producerId &&
          !isVerificationRevoked(att, v.id) &&
          !isVerificationExpired(v)
      )
  );
};
```

Reply:
I agree, it would be better to refactor it since there are a lot of checks